ttnn/cpp/ttnn/operations/data_movement/repeat/repeat.cpp

```diff
@@ -18,10 +21,18 @@ ttnn::Tensor RepeatOperation::invoke(
     const Shape & repeat_dims,
     const std::optional<MemoryConfig>& memory_config_arg) {
 
-    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    // special path for handling padded tiled tensors that are naively repeated
+    // *with padding* in the current implementation
+    std::optional<Tensor> formatted_input_tensor = std::nullopt;
+    if (input_tensor.get_layout() != Layout::ROW_MAJOR
+        && input_tensor.get_legacy_shape().without_padding() != input_tensor.get_legacy_shape()) {
+        formatted_input_tensor = ttnn::untilize(input_tensor);
```

Comment:
Curious to learn, whether it's possible to fix the issue without switching to ROW_MAJOR and back. 
Can we change `RepeatDeviceOperation` to adapt for TILE case? What do you think?

Reply:
We'll address this after the functional generality efforts if the current implementation is deemed insufficiently performant.